### ISSUE TITLE
taskotron results don't show up for all tested architectures

### ISSUE THREAD

**Issue by kparal** [fedora-infra/bodhi#2425]
We have recently done a few changes in how taskotron tests are scheduled and reported. We now run tests on both `x86_64` and `armhfp` builds/repos, and we often execute this separately (two runs), instead all in a single run. This was to increase tests reliability and allow future scaling. As a consequence, results shown in Bodhi got a bit out of sync, and don't work as expected. I believe this affects just `dist.rpmdeplint`, `dist.abicheck` and `dist.rpmlint`, where these changes have been performed.  For these tests, Bodhi shows up the very last report in ResultsDB, regardless of architecture. That means you sometimes see `dist.rpmdeplint` result for `x86_64`, and sometimes for `armhfp`, based on which one was executed last. The same applies to the other two tests. The ideal state would of course be to show results for all (both) architectures.

I was the one writing the taskotron querying code in the past, but I didn't have time to delve deep into it again, so I'm operating mostly from memory (sorry if I get something wrong). As a reference, I'll use:
https://bodhi.fedoraproject.org/updates/FEDORA-2018-25ccbfaf95

Seeing which queries Bodhi javascript makes, I think this could be solved pretty easily by adjusting this request:
```
latest?item=$item&type=koji_build&testcases:like=dist.*
```
so in this particular example this one:
https://taskotron.fedoraproject.org/resultsdb_api//api/v2.0/results/latest?item=vim-8.1.035-1.fc28&type=koji_build&testcases:like=dist.*

into two separate requests with a specified architecture:
```
latest?item=$item&type=koji_build&testcases:like=dist.*&arch=x86_64
latest?item=$item&type=koji_build&testcases:like=dist.*&arch=armhfp
```
i.e. these two:
https://taskotron.fedoraproject.org/resultsdb_api//api/v2.0/results/latest?item=vim-8.1.035-1.fc28&type=koji_build&testcases:like=dist.*&arch=x86_64
https://taskotron.fedoraproject.org/resultsdb_api//api/v2.0/results/latest?item=vim-8.1.035-1.fc28&type=koji_build&testcases:like=dist.*&arch=armhfp

And then making sure the results show up properly in the web UI (sorted, etc). Please note that some of the results will be part of both queries (tests which still run on both architectures in a single run, for different reasons, report the result marked with both arches - that includes `dist.rpmgrill` and `dist.python-versions`, iirc), so you'll need to deduplicate those. Should be simple since all results have an unique ID.

I'm not sure if the queries performed by the server side of Bodhi (python, not javascript) also need to be adjusted, or whether all of that code has been moved to greenwave.

@bowlofeggs If you prefer, @lukassparrow was willing to look into this.

**Comment by bowlofeggs** [comment]
Hello @kparal!

@ralphbean has an open pull request in which he is having Bodhi live query Greenwave in the JavaScript, and it seems to me that his patch gets the results from Greenwave instead of directly from ResultsDB:

https://github.com/fedora-infra/bodhi/pull/2393

So I have a few thoughts:

0. Does Greenwave do the right thing WRT to your concerns above when it queries ResultsDB?
1. I'd love your feedback on that PR if you have thoughts about it.
2. If the answer to 0 is yes, should we just mark that PR as resolving this issue?

**Comment by kparal** [comment]
I'm going for a vacation for a week, @lukassparrow can you please look at greenwave and the PR and post your thoughts? I guess you could apply that PR in your local bodhi instance and compare how the results render.

In general (without reading that PR in detail), if Bodhi is going to simply ask Greenwave "give me everything important for build X", then this ticket doesn't apply, yes. Instead we'll need to have a look at Greenwave and submit potential issues against that.

**Comment by lbrabec** [comment]
Local instance of Bodhi with PR #2393 show separate results for `x86_64` and `armhfp`, so the PR seems to resolve this issue.

**Comment by bowlofeggs** [comment]
A fix for this issue is planned to be included in the upcoming 3.9.0 release: #2477.

**Comment by bowlofeggs** [comment]
The patch for this issue was deployed to Fedora's production Bodhi instance yesterday.
